addons/udes_stock/models/stock_location.py

```diff
@@ -106,9 +104,51 @@ def _domain_height_category(self):
 
                 If not set then the location will use the format
                 of the nearest ancestor that has a format specified.
-                """
+                """,
+    )
+    u_state = fields.Selection(
+        selection=[
+            ("empty", "Empty"),
+            ("blocked", "Blocked"),
+            ("has_stock", "Has Stock"),
+            ("archived", "Archived"),
+        ],
+        compute="_compute_state",
+        store=True,
+        string="State",
+        help="""Computed field describing the state of the stock location.""",
     )
 
+    @api.depends("quant_ids", "usage", "active")
```

Comment:
Sorry something that wasn't considered while making the story, we'll also need `u_is_countable` and `u_location_is_countable` - so that we cut down on locations that have this computed. Can you also port these two fields?